.github/workflows/build_and_test_dbt.yaml

```diff
@@ -23,35 +23,20 @@ jobs:
         with:
           role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
 
-      # We have to use the separate `restore`/`save` actions instead of the
-      # unified `cache` action because only `restore` provides access to the
-      # `cache-matched-key` and `cache-primary-key` outputs as of v3
       - name: Restore dbt state cache
         id: cache
-        uses: actions/cache/restore@v3
+        uses: ./.github/actions/restore_dbt_cache
         with:
           path: ${{ env.PROJECT_DIR }}/${{ env.STATE_DIR }}
           key: ${{ env.CACHE_NAME }}-${{ env.CACHE_KEY }}
-          restore-keys: |
-            ${{ env.CACHE_NAME }}-master
+          restore-key: ${{ env.CACHE_NAME }}-master
 
-      # If we restore the cache from the `restore-keys` key, the `cache-hit`
-      # output will be 'false' but the `cache-matched-key` output will be
-      # the name of the `restore-keys` key; we want to count this case as a hit
-      - if: |
-          steps.cache.outputs.cache-hit == 'true' ||
-          steps.cache.outputs.cache-matched-key == format(
-            '{0}-master', env.CACHE_NAME
-          )
+      - if: steps.cache.outputs.cache-hit == 'true'
```

Comment:
One nice benefit of this change is that we can implement a definition of `cache-hit` that makes more sense to me (i.e. it's still a hit even if we restore from the fallback cache)!